test/JsonApiDotNetCoreMongoDbExampleTests/IntegrationTestContext.cs

```diff
@@ -0,0 +1,227 @@
+                }
+            }
+
+            using HttpClient client = Factory.CreateClient();
+
+            if (acceptHeaders != null)
+            {
+                foreach (var acceptHeader in acceptHeaders)
+                {
+                    client.DefaultRequestHeaders.Accept.Add(acceptHeader);
+                }
+            }
+
+            HttpResponseMessage responseMessage = await client.SendAsync(request);
+
+            string responseText = await responseMessage.Content.ReadAsStringAsync();
+            var responseDocument = DeserializeResponse<TResponseDocument>(responseText);
+
+            return (responseMessage, responseDocument);
+        }
+
+        private string SerializeRequest(object requestBody)
+        {
+            return requestBody == null
+                ? null
+                : requestBody is string stringRequestBody
+                    ? stringRequestBody
+                    : JsonConvert.SerializeObject(requestBody);
+        }
+
+        private TResponseDocument DeserializeResponse<TResponseDocument>(string responseText)
+        {
+            if (typeof(TResponseDocument) == typeof(string))
+            {
+                return (TResponseDocument)(object)responseText;
+            }
+
+            try
+            {
+                return JsonConvert.DeserializeObject<TResponseDocument>(responseText);
+            }
+            catch (JsonException exception)
+            {
+                throw new FormatException($"Failed to deserialize response body to JSON:\n{responseText}", exception);
+            }
+        }
+        
+        private sealed class IntegrationTestWebApplicationFactory : WebApplicationFactory<EmptyStartup>
+        {
+            private Action<IServiceCollection> _beforeServicesConfiguration;
+            private Action<IServiceCollection> _afterServicesConfiguration;
+
+            public void ConfigureServicesBeforeStartup(Action<IServiceCollection> servicesConfiguration)
+            {
+                _beforeServicesConfiguration = servicesConfiguration;
+            }
+
+            public void ConfigureServicesAfterStartup(Action<IServiceCollection> servicesConfiguration)
+            {
+                _afterServicesConfiguration = servicesConfiguration;
+            }
+
+            protected override IHostBuilder CreateHostBuilder()
+            {
+                return Host.CreateDefaultBuilder(null)
+                    .ConfigureWebHostDefaults(webBuilder =>
+                    {
+                        webBuilder.ConfigureTestServices(services =>
+                        {
+                            _beforeServicesConfiguration?.Invoke(services);
+                        });
+                    
+                        webBuilder.UseStartup<TStartup>();
+
+                        webBuilder.ConfigureTestServices(services =>
+                        {
```

Comment:
Not that I can remember, I looked at the implementation of `IntegrationTestContext` in the original repo and left it like it was there.

Reply:
That's strange. Looking at the [diff](https://github.com/json-api-dotnet/JsonApiDotNetCore/commits/master/test/JsonApiDotNetCoreExampleTests/IntegrationTestContext.cs), it has always been `ConfigureServices` since I added this file last summer.